pkg/upstream/rewrite.go

```diff
@@ -0,0 +1,40 @@
+package upstream
+
+import (
+	"net/http"
+	"net/url"
+	"regexp"
+
+	"github.com/justinas/alice"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/logger"
+)
+
+// newRewritePath creates a new middleware that will rewrite the request URI
+// path before handing the request to the next server.
+func newRewritePath(rewriteRegExp *regexp.Regexp, rewriteTarget string) alice.Constructor {
+	return func(next http.Handler) http.Handler {
+		return rewritePath(rewriteRegExp, rewriteTarget, next)
+	}
+}
+
+// rewritePath uses the regexp to rewrite the request URI based on the provided
+// rewriteTarget.
+func rewritePath(rewriteRegExp *regexp.Regexp, rewriteTarget string, next http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+		reqURL, err := url.Parse(req.RequestURI)
```

Comment:
is it really necessary to re-parse the original request? woudn't it suffice to operate on the existing `req.URL` instance (maybe clone it to avoid side effects elsewhere)

Reply:
I think there are other parts of the code that interfere with it before we get here (will double check), but as I was working on this I thought it would be safer to just ensure we use the original request URI